CONTRIBUTING.md

```diff
@@ -0,0 +1,368 @@
+* We are always looking to help, please feel free to open an [issue](https://github.com/aws-amplify/aws-sdk-android/issues).
+
+## Workflows
+
+#### Modifying code for a new feature or bug fix
+
+Be aware of the Getting Started and Pull Request guides. This portion deals with some guidelines with regards to actually coding a feature/bug fix.
+
+In case of new features, you should proceed by creating your feature in
+isolation. This way, you can write independent unit tests for your feature.
+Unit tests are required for all features being added to the code-base. Once you
+have the skeleton of your feature working, you can integrate it into the
+existing source files.
+
+Writing good Android code and good tests is far outside the scope of this
+document. But, here are some of the conventions and expectations for source files in the Android codebase.
+
+* All files get a copyright header with the year of creation;
+```java
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+```
+* Use [`@NonNull`](https://developer.android.com/reference/kotlin/androidx/annotation/NonNull)
+  and [`@Nullable`](https://developer.android.com/reference/kotlin/androidx/annotation/Nullable)
+  annotations on any/all params, so that
+  [Kotlin knows what to do](https://kotlinlang.org/docs/reference/java-interop.html#nullability-annotations),
+  when it consumes the library;
+* Label Arrange/Act/Assert portions of your unit tests, so others can follow
+  their flow;
+* Arrange shared dependencies and test objects in an
+  [`@Before`](http://junit.sourceforge.net/javadoc/org/junit/Before.html) method;
+* Prefer `private` and `final` class members where-ever possible. This limits scope
+  and mutability;
+* Add documentation to anything that says `public` or `protected`. The
+  checkstyle doesn't require documentation on package-local or `private` fields, but
+  that might be a good idea, too, if you're doing some interesting/non-trivial
+  work in those places.
+
+These are just some general guidelines, but
+[Checkstyle](https://github.com/checkstyle/checkstyle/blob/master/README.md) and
+[Android Lint](https://developer.android.com/studio/write/lint#overview)
+are the ultimate authorities on this, and will tell you if something is
+badly formatted, or exciting a known shortcoming in Java, Android, or
+their library ecosystem.
```

Comment:
The SDK does not use checkstyle, and does not use Lint in any uniform way. We should remove this paragraph. This paragraph makes it sounds like these tools are a big deal in the SDK. Which, unfortunately, they are not, currently.